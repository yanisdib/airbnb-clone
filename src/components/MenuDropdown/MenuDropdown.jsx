import styled from 'styled-components';

import { Dropdown } from '..';


const Wrapper = styled.div`
    width: 100%;
    height: 100%;
    margin-left: 0 !important;
`;

const AuthLinkList = styled.ul`
    padding: 10px 0;
    border-bottom: 1px solid #DFDFDF;

    a {
        display: flex;
        width: fit-content;
        color: #000000;
        font-size: 0.9em;
        font-weight: 400;
        padding: 10px 16px;
        text-decoration: none;

        &:first-child {
            font-weight: 500;
        }
    }
`;

const PublicLinkList = styled.ul`
    padding: 10px 0;

    a {
        display: flex;
        width: fit-content;
        color: #000000;
        font-size: 0.9em;
        font-weight: 400;
        padding: 10px 16px;
        text-decoration: none;

        &:first-child {
            font-weight: 500;
        }
    }
`;


function MenuDropdown() {
    return (
        <Dropdown width={255}>
            <Wrapper>
                <AuthLinkList>
                    <a href="/">Sign up</a>
                    <a href="/">Log in</a>
                </AuthLinkList>
                <PublicLinkList>
                    <a href="/">Host your home</a>
                    <a href="/">Host an experience</a>
                    <a href="/">Help</a>
                </PublicLinkList>
            </Wrapper>
        </Dropdown>
    );
}


export default MenuDropdown;