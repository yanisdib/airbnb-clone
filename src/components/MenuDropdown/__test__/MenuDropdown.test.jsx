import { shallow } from 'enzyme';

import MenuDropdown from '../MenuDropdown';


describe('MenuDropdown component', () => {
    let wrapper;

    beforeEach(() => wrapper = shallow(<MenuDropdown />))

    it('renders correctly without crashing', () => expect(wrapper).toMatchSnapshot())
})