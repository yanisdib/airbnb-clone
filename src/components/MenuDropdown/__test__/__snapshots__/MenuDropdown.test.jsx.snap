// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuDropdown component renders correctly without crashing 1`] = `
<styled.div>
  <styled.div>
    <styled.ul>
      <a
        href="/"
      >
        Sign up
      </a>
      <a
        href="/"
      >
        Log in
      </a>
    </styled.ul>
    <styled.ul>
      <a
        href="/"
      >
        Host your home
      </a>
      <a
        href="/"
      >
        Host an experience
      </a>
      <a
        href="/"
      >
        Help
      </a>
    </styled.ul>
  </styled.div>
</styled.div>
`;
